public/app/plugins/datasource/azuremonitor/components/ResourcePicker/ResourcePicker.tsx

```diff
@@ -85,7 +88,21 @@ const ResourcePicker = ({
     const subscriptions = await datasource.getSubscriptions();
     setSubscriptions(subscriptions.map((sub) => ({ label: sub.text, value: sub.value })));
     setIsLoadingSubscriptions(false);
-  }, [datasource]);
+
+    if (queryType === 'metrics') {
+      setIsLoadingNamespaces(true);
+      const namespaces = await datasource.getMetricNamespaces(
+        subscriptions[0]?.value || datasource.getDefaultSubscriptionId()
+      );
+      setNamespaces(
+        namespaces?.map((ns) => ({
```

Comment:
if `namespaces` is `null` or `undefined` for any reason, this will set `namespaces` back to `undefined`. 

I think that `tsc` didn't catch this as `namespaces` is defined twice in the same scope (L73, L94), and it's not clear which one is written here.

Reply:
I've updated the variable name for clarity but I don't believe `namespaces` can be `null` or `undefined` as `getMetricNamespaces` should return an empty array in both the failure or success case (if there are no namespaces available).